storage/operation/receipts.go

```diff
@@ -24,17 +29,29 @@ func RetrieveExecutionReceiptStub(r storage.Reader, receiptID flow.Identifier, m
 
 // IndexOwnExecutionReceipt indexes the Execution Node's OWN execution receipt by the executed block ID.
 //
-// CAUTION:
-//   - OVERWRITES existing data (potential for data corruption):
-//     This method silently overrides existing data without any sanity checks whether data for the same key already exits.
-//     Note that the Flow protocol mandates that for a previously persisted key, the data is never changed to a different
-//     value. Changing data could cause the node to publish inconsistent data and to be slashed, or the protocol to be
-//     compromised as a whole. This method does not contain any safeguards to prevent such data corruption. The caller
-//     is responsible to ensure that the DEDUPLICATION CHECK is done elsewhere ATOMICALLY with this write operation.
-//
-// No errors are expected during normal operation.
-func IndexOwnExecutionReceipt(w storage.Writer, blockID flow.Identifier, receiptID flow.Identifier) error {
-	return UpsertByKey(w, MakePrefix(codeOwnBlockReceipt, blockID), receiptID)
+// Error returns:
+//   - [storage.ErrDataMismatch] if a *different* receipt has already been indexed for the same block
+func IndexOwnExecutionReceipt(lctx lockctx.Proof, rw storage.ReaderBatchWriter, blockID flow.Identifier, receiptID flow.Identifier) error {
```

Comment:
```suggestion
func IndexMyExecutionReceipt(lctx lockctx.Proof, rw storage.ReaderBatchWriter, blockID flow.Identifier, receiptID flow.Identifier) error {
```
Bit of a nitpick. I think "my" is clearer, since "own" is also a common verb. Feel free to ignore if you disagree.